pkg/cmd/server/origin/master.go

```diff
@@ -193,7 +193,28 @@ func (c *MasterConfig) Run(kc *kubernetes.MasterConfig, assetConfig *AssetConfig
 	for _, s := range messages {
 		glog.Infof(s, c.Options.ServingInfo.BindAddress)
 	}
-	go kmaster.GenericAPIServer.PrepareRun().Run(utilwait.NeverStop)
+
+	apiserver := kmaster.GenericAPIServer.PrepareRun()
+
+	// presence of the key indicates whether or not to enable the aggregator
+	if len(c.Options.AggregatorConfig.ProxyClientInfo.KeyFile) == 0 {
+		go apiserver.Run(utilwait.NeverStop)
+
+		// Attempt to verify the server came up for 20 seconds (100 tries * 100ms, 100ms timeout per try)
+		cmdutil.WaitForSuccessfulDial(c.TLS, c.Options.ServingInfo.BindNetwork, c.Options.ServingInfo.BindAddress, 100*time.Millisecond, 100*time.Millisecond, 100)
```

Comment:
> pre-existing. Not really sure what it does

Tries a bunch of times and returns and error on failure (which we ignore and just keep going).